### PR TITLE
feat: split backend processes scenarios into a new feature file

### LINKED ISSUE #530: [Fleet] Split off Metricbeat and Filebeat based Agent assessments into separate files (like we do for Endpoint)
This would provide:
 - better parallelization (which will matter if / when we have more tests)
 - easier assessment of Docs & communications in ES / Fleet Server & Kibana

I propose that with current support added to the framework building blocks, we can now easily set policy to be what we want and exclude what we don't want so we can assess the individual Beats.

Came up out of a discussion with PH.

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It extracts all processes-related scenarios from `Fleet Mode` feature file to a new feature file. In this new file, we will namespace all scenarios under the `@backend_processes` Cucumber annotation, so that we will be able to add a parallel branch for the test suite.

To accomplish it, we copied the scenarios from Fleet to the new file, removing all steps not related to processes. And, in the fleet file, we removed any process-related step, keeping functional steps in there.

## Why is it important?
We expect to two things:

1. reduce the CI build, as all the wait-for-process logic will be executed in the new feature file (in a parallel CI branch)
1. decouple errors related to backend process from other type of errors

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests for the CLI, and they are passing locally
- [x] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [ ]

## How to test this PR locally

```shell
SUITE="fleet" TAGS="backend_processes" TIMEOUT_FACTOR=3 LOG_LEVEL=TRACE DEVELOPER_MODE=true make -C e2e functional-test 
```

## Related issues

- Resolves #530

## Follow-ups
Measure CI time for both the `backend_processes` and the `fleet_agent_mode` parallel branches.

### CODE REVIEW

**Comment by mdelapenya** [comment]
jenkins run the tests please

File: .ci/.e2e-tests.yaml [hunk 1]
```diff
@@ -6,9 +6,9 @@ SUITES:
     scenarios:
       - name: "APM Server"
         tags: "apm-server"
-      - "Filebeat":
+      - name: "Filebeat"
         tags: "filebeat"
-      - "Metricbeat":
+      - name: "Metricbeat"
```
Comment (mdelapenya) [inline .ci/.e2e-tests.yaml:11]: Consistency: cosmetic change

File: .ci/.e2e-tests.yaml [hunk 2]
```diff
@@ -20,6 +20,8 @@ SUITES:
         tags: "agent_endpoint_integration"
       - name: "Stand-alone"
         tags: "stand_alone_agent"
+      - name: "Backend Processes"
+        tags: "backend_processes"
```
Comment (mdelapenya) [inline .ci/.e2e-tests.yaml:24]: This will create a new parallel branch for the new tag

**Review by cachedout — APPROVED** [review]

**Review by michalpristas — APPROVED** [review]
makes sense

**Pull request merged**